app/controllers/foreman_azure_rm/concerns/compute_resources_controller_extensions.rb

```diff
@@ -3,6 +3,15 @@ module Concerns
     module ComputeResourcesControllerExtensions
       include Api::Version2
       include Foreman::Controller::Parameters::ComputeResource
+      extend ::Apipie::DSL::Concern
+      update_api(:create, :update) do
+        param :compute_resource, Hash do
+          # Not adding :tenant as already specified in core.
+          param :app_ident, String, :desc => N_("Client ID for AzureRM")
+          param :secret_key, String, :desc => N_("Client Secret for AzureRM")
+          param :sub_id, String, :desc => N_("Subscription ID for AzureRM")
+        end
```

Comment:
I suppose we will need to add other fields specified in https://github.com/theforeman/foreman_azure_rm/pull/43

Reply:
You mean fields such as `vm_size`, `username`, `password` etc. in azure_rm_compute.rb ? 
But, I took reference from [kubevirt](https://github.com/theforeman/foreman_kubevirt/blob/99e87c872f3d2aa4f3fa9b7c8593a31a72ea689f/app/controllers/foreman_kubevirt/concerns/api/compute_resources_controller_extensions.rb) which only specifies fields for creating a compute resource.